src/RealtimeChannel.ts

```diff
@@ -395,6 +395,15 @@ export default class RealtimeChannel {
       [key: string]: any
     }) => void
   ): RealtimeChannel
+  on<T extends { [key: string]: any }>(
+    type: `${REALTIME_LISTEN_TYPES.BROADCAST}`,
+    filter: { event: string },
+    callback: (payload: {
+      type: `${REALTIME_LISTEN_TYPES.BROADCAST}`
+      event: string
+      [key: string]: any
```

Comment:
@filipecabaco was there a reason you kept the following above:

```ts
on(
    type: `${REALTIME_LISTEN_TYPES.BROADCAST}`,
    filter: { event: string },
    callback: (payload: {
      type: `${REALTIME_LISTEN_TYPES.BROADCAST}`
      event: string
      [key: string]: any
    }) => void
  ): RealtimeChannel
```

and didn't replace it with this?

also, you'll want to use the generic you introduced:

```suggestion
      payload: T
```

Reply:
great call! changing